go/enclave/components/rollup_consumer.go

```diff
@@ -119,58 +119,60 @@ func (rc *rollupConsumerImpl) processRollups(br *common.BlockAndReceipts) ([]*co
 	}
 
 	if len(rollups) == 0 {
-		return nil, nil
+		return nil, nil //nolint:nilnil
 	}
 
-	validRollups := make([]*core.Rollup, 0)
-
+	var signedRollup *core.Rollup
 	blockHash := block.Hash()
+
+	// loop through the rollups, find the one that is signed, verify the signature, make sure it's the only one
 	for _, rollup := range rollups {
 		if err = rc.verifier.CheckSequencerSignature(rollup.Hash(), &rollup.Header.Agg, rollup.Header.R, rollup.Header.S); err != nil {
 			return nil, fmt.Errorf("rollup signature was invalid. Cause: %w", err)
 		}
-
-		if !rollup.IsGenesis() {
-			if err = rc.checkRollupsCorrectlyChained(rollup, latestRollup); err != nil {
-				rc.logger.Warn("Rollup was not correctly chained",
-					"height", rollup.NumberU64(), "hash", rollup.Hash(), log.ErrKey, err)
-				// we need to check the other rollups in this block still otherwise a correct one could get missed
-				continue
-			}
+		if signedRollup != nil {
+			// todo (@matt) - make sure this can't be used to DOS the network
+			// we should never receive multiple signed rollups in a single block, the host should only ever publish one
+			return nil, fmt.Errorf("received multiple signed rollups in single block %s", blockHash)
 		}
+		signedRollup = rollup
+	}
+	if signedRollup == nil {
+		return nil, nil //nolint:nilnil
+	}
 
-		// todo (@matt) - store batches from the rollup (important during catch-up)
-
-		// rollup has been verified as the next valid rollup. If there are more rollups to process then we build then on top of this one
-		validRollups = append(validRollups, rollup)
-		latestRollup = rollup
-		if err = rc.storage.StoreRollup(rollup); err != nil {
-			// todo (@matt) - this seems catastrophic, how do we recover the lost rollup in this case?
-			return nil, fmt.Errorf("could not store rollup. Cause: %w", err)
-		}
+	if err = rc.checkRollupsCorrectlyChained(signedRollup, latestRollup); err != nil {
+		rc.logger.Warn("Rollup was not correctly chained",
+			"height", signedRollup.NumberU64(), "hash", signedRollup.Hash(), log.ErrKey, err)
```

Comment:
Shouldn't it return in this branch?
What's the point in continuing if the rollup was incorrect?

Reply:
Yeah good catch, that was left over from the loop